lib/puppet_x/puppetlabs/scheduled_task/taskscheduler2_task.rb

```diff
@@ -31,7 +31,7 @@ def initialize(work_item = nil, trigger = nil)
   def enum
     @tasksched.enum_task_names(PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::ROOT_FOLDER,
       include_child_folders: false,
-      include_compatibility: [PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_COMPATIBILITY_V2]).map do |item|
+      include_compatibility: [PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_COMPATIBILITY_V2, PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_COMPATIBILITY_V1]).map do |item|
```

Comment:
Was investigating having this return all available tasks instead of just V2, and therefore allowing `self.instances` in the provider to correctly populate with all manageable scheduled tasks on the system.

Reply:
Yeah, we should discuss further, but in the new PR I just replaced `TaskScheduler2V1Task` with `TaskScheduler2Task` everywhere, and I added `AT` in addition to `V1` like you've done here.

That completely removes the adapter from the workflow, which now makes me wonder why we needed `TaskScheduler2Task` class in the first place. I think removing all the Trigger related code has made the picture a lot clearer now.